provider/openstack/firewaller.go

```diff
@@ -545,92 +545,139 @@ func (c *neutronFirewaller) setUpGlobalGroup(groupName string, apiPort int) (neu
 // zeroGroup holds the zero security group.
 var zeroGroup neutron.SecurityGroupV2
 
-// ensureGroup returns the security group with name and perms.
+// ensureGroup returns the security group with name and rules.
 // If a group with name does not exist, one will be created.
-// If it exists, its permissions are set to perms.
+// If it exists, its permissions are set to rules.
 func (c *neutronFirewaller) ensureGroup(name string, rules []neutron.RuleInfoV2) (neutron.SecurityGroupV2, error) {
 	neutronClient := c.environ.neutron()
+	var group neutron.SecurityGroupV2
+
 	// First attempt to look up an existing group by name.
 	groupsFound, err := neutronClient.SecurityGroupByNameV2(name)
-	if err == nil {
-		for _, group := range groupsFound {
-			if c.verifyGroupRules(group.Rules, rules) {
-				return group, nil
-			}
+	// a list is returned, but there should be only one
+	if err == nil && len(groupsFound) == 1 {
+		group = groupsFound[0]
+	} else if len(groupsFound) == 0 {
+		// A SecurityGroup with the requested name doesn't exist,
+		// create it.  err will say name not found, so ignore.
```

Comment:
It's not sufficient to ignore the error. The could be some other issue like a connectivity or auth failure etc. We need to check that the error is a not found type of error, otherwise exit wit the error.